drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/compiler/OCLCompilationResultBuilder.java

```diff
@@ -236,23 +263,6 @@ void emitBlock(Block block) {
         printBasicBlockTrace(block);
 
         LIRInstruction breakInst = null;
-        LIRInstruction opPreEmit = null;
-
-        // SchedulePhase.SchedulingStrategy.LATEST_OUT_OF_LOOPS in the latest Graal
-        // reschedule unreachable within the loop instruction to the previous basic
-        // block (i.e. the one that the loop begin exists). This patch solves the issue
-        // of Loop header BBs that contain additional ops
-        for (int i = 0; i < lir.getLIRforBlock(block).size(); i++) {
-            if (isLoopDependencyNode(lir.getLIRforBlock(block).get(i))) {
-                for (int j = i; j < lir.getLIRforBlock(block).size(); j++) {
-                    if (!isLoopDependencyNode(lir.getLIRforBlock(block).get(j))) {
-                        emitOp(this, lir.getLIRforBlock(block).get(j));
-                        opPreEmit = lir.getLIRforBlock(block).get(j);
-                    }
-                }
-                break;
-            }
-        }
```

Comment:
Why is this part removed?

Reply:
This part tried to fix a similar issue but was only moving a single instruction outside the loop. 
I think that my fix covers this case